### PR TITLE
Move default-settings to /etc/xpra

### PR DESCRIPTION
**Pull request by basilgello**
The default-settings.txt file is a configuration file
editable by user, so its place on Unix systems is
in /etc.

Also fix typo in make_rpm

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -513,6 +524,17 @@ def main(args):
         if len(args)>=4:
             minifier = args[3]
         install_html5(install_dir, minifier)
+        if len(args)<3:
```
Comment (totaam) [inline setup.py:527]: Why only if there are less than 3 args?
Shouldn't this be done in all cases?

File: setup.py [hunk 2]
```diff
@@ -426,6 +426,17 @@ def make_deb():
     os.mkdir("./xpra-html5")
     shutil.copytree("./packaging/debian", "./xpra-html5/DEBIAN")
     install_html5("./xpra-html5/usr/share/xpra/www/", "uglifyjs")
+    # Move and symlink default-settings.txt
+    os.makedirs("./xpra-html5/etc/xpra/html5-client", exist_ok=True)
+    default_settings_files = { "default-settings.txt", "default-settings.txt.gz", "default-settings.txt.br" }
+    for filename in default_settings_files:
```
Comment (totaam) [inline setup.py:432]: I don't like having 3 copies of what is effectively the same file in `/etc`
We should just prevent this file from ever being minified instead.

File: setup.py [hunk 3]
```diff
@@ -426,6 +426,17 @@ def make_deb():
...
+        print("Checking existence of %s" % os.path.join("./xpra-html5/usr/share/xpra/www", filename))
+        if os.path.exists(os.path.join("./xpra-html5/usr/share/xpra/www", filename)):
+           os.rename(os.path.join("./xpra-html5/usr/share/xpra/www", filename),
```
Comment (totaam) [inline setup.py:435]: Instead of repeating the same path 4 times, please use a variable to make the code more readable. ie: `www_dir`.

File: setup.py [hunk 4]
```diff
@@ -426,6 +426,17 @@ def make_deb():
...
+               os.path.join("./xpra-html5/etc/xpra/html5-client", filename))
+           os.symlink(os.path.join("/etc/xpra/html5-client", filename),
+               os.path.join("./xpra-html5/usr/share/xpra/www", filename))
```
Comment (totaam) [inline setup.py:438]: Why not just symlink this file from `www_dir` directly to `/etc/xpra/html5-client/`?
(and since it should be just one file - is the `html5-client` directory even needed?)

File: setup.py [hunk 5]
```diff
@@ -513,6 +524,17 @@ def main(args):
         if len(args)>=4:
             minifier = args[3]
         install_html5(install_dir, minifier)
+        if len(args)<3:
+            # Move and symlink default-settings.txt
+            os.makedirs(os.path.normpath(os.path.join(sys.prefix, "../etc/xpra/html5-client")), exist_ok=True)
+            default_settings_files = { "default-settings.txt", "default-settings.txt.gz", "default-settings.txt.br" }
```
Comment (totaam) [inline setup.py:530]: As above.
Simply not having the `.gz` and `.br` is cleaner.

**Review by totaam — CHANGES_REQUESTED** [review]
Sorry, I had forgotten about this PR.

File: setup.py (hunk 2 above)
Comment (basilgello) [reply setup.py:432]: Yes, I will fix it!

File: setup.py (hunk 3 above)
Comment (basilgello) [reply setup.py:435]: This will be fixed too

File: setup.py (hunk 4 above)
Comment (basilgello) [reply setup.py:438]: The idea of whole change os that configuration files must reside in sub-directories of `/etc`, so symlink in `/usr/share/xpra/www` points to editable `/etc/xpra/default-settings.txt`. Separate directory for `html5-client` is not needed if we rename the file like `html5-client-settings.txt`.

File: setup.py (hunk 5 above)
Comment (basilgello) [reply setup.py:530]: Will fix

File: setup.py (hunk 1 above)
Comment (basilgello) [reply setup.py:527]: I recall I did it to preserve `default-settings.txt` inside `install_dir` if the installation directory is specified by user. This should keep the logic for `bdist` intact.

**Comment by basilgello** [comment]
@totaam The new revision ships only `default-settings.txt`. I decided (for now) to ship it in `/etc/xpra/html5-client/` because we need either to rename file in `/etc/xpra/` to something like `html5-client-settings.txt` or to add migration code for existing `default-settings.txt` files installed without package managers like `rpm` or `dpkg`.

File: packaging/debian/changelog [hunk 6]
```diff
@@ -1,4 +1,4 @@
-xpra-html5 (4.5.2-r1075-1) UNRELEASED; urgency=low
+xpra-html5 (4.5.2-r214-1) UNRELEASED; urgency=low
```
Comment (totaam) [inline packaging/debian/changelog:1]: Please remove these changes from the PR.
(it's unrelated and for some reason, it's going backwards?)

File: setup.py [hunk 7]
```diff
@@ -16,6 +16,11 @@
 AUTHOR = "Antoine Martin"
 AUTHOR_EMAIL = "antoine@xpra.org"
 
+# List of configuration files
+# These must not be minified or compressed and must be moved to /etc/xpra
+# and symlinked to /usr/share/xpra/www if no custom installation directory
+# is set
+configuration_files = { "default-settings.txt" }
```
Comment (totaam) [inline setup.py:23]: You say this is a _List of configuration files_ but it is actually a set...
Then later on, you call it a `not_minify_dict`, but it isn't a dict.

File: packaging/debian/changelog (hunk 6 above)
Comment (basilgello) [reply packaging/debian/changelog:1]: Oops, it is an artifact from testing the build! Sorry! :)

File: setup.py (hunk 7 above)
Comment (basilgello) [reply setup.py:23]: Let's settle with `Configuration files that must not be minified and compressed…` and `not_minify_files` then?

Comment (totaam) [reply setup.py:23]: OK!
I would use an array: `configuration_files = ["default-settings.txt"]` and `not_minify_files`.

File: setup.py [hunk 8]
```diff
@@ -456,7 +475,7 @@ def make_rpm():
     SOURCES = RPMBUILD+"/SOURCES"
     if not os.path.exists(SOURCES):
         os.makedirs(SOURCES, 0o755)
-    os.rename("dist/" + tarxz, SOURCES + tarxz)
+    os.rename("dist/" + tarxz, os.path.join(SOURCES, tarxz))
```
Comment (totaam) [inline setup.py:478]: This change is unrelated,  I've rolled it into: 0c3ba3b777673846ad377211975c9bd50f1a3ea3

File: setup.py (hunk 7 above)
Comment (basilgello) [reply setup.py:22]: OK! Then let me push onve more ;)

File: setup.py (hunk 8 above)
Comment (basilgello) [reply setup.py]: Done!

**Comment by totaam** [comment]
Just noticed:
```
conf_dir = os.path.normpath(os.path.join(sys.prefix, "../etc/xpra/html5-client"))
```
That won't work for `sys.prefix = "/usr/local"` as found on FreeBSD:
```
> os.path.join("/usr/local", "../etc/xpra/html5-client")
'/usr/local/../etc/xpra/html5-client'
```

And the RPMs don't have `/etc/xpra/html5-client`. Can you remind me why the logic is both in `if cmd=="install":` and `make_deb`? Why not just roll it into `install_html5` just once?

I will also have to ensure this doesn't cause problems for the MS Windows and MacOS builds...

**Comment by basilgello** [comment]
> Can you remind me why the logic… 

There are different paths. make_deb operates from `./xpra-html5` as sysroot while install operates on `/`

> as found on FreeBSD:

Just use `../../` instead?

**Comment by basilgello** [comment]
> the MS Windows and MacOS builds

MS Windows is not POSIX, macos is posix but it has `/etc`.

**Comment by totaam** [comment]
> MS Windows is not POSIX

Our builds are based on MSYS2 (mingw64) so they are actually a lot more posixy than you would think.
And we do ship an `/etc` directory in the application image. But since there is no proper symlink support and `/etc` is not in a user editable location. (although there are other user-editable locations, like the ones we use for the ssl certs)
Anyway let's not bother.

> macos is posix but it has `/etc`.

Not one we can use.
Again, there are other locations, but let's not bother.

**Comment by totaam** [comment]
> Just use `../../` instead?

How?

**Comment by basilgello** [comment]
`conf_dir = os.path.normpath(os.path.join(sys.prefix, "../../etc/xpra/html5-client"))` and wrap it to not win32 ?

**Comment by basilgello** [comment]
Then let me refactor this into a sysroot helper?
